src/oic/oauth2/base.py

```diff
@@ -128,7 +128,10 @@ def http_request(self, url, method="GET", **kwargs):
             _kwargs = self.req_callback(method, url, **_kwargs)
 
         try:
-            r = requests.request(method, url, **_kwargs)  # type: ignore
+            if self.settings.request_session:
```

Comment:
```suggestion
            if self.settings.request_session is not None:
```
Testing for `None` is better explicit.

Reply:
Replaced with `if getattr(self.settings, "request_session", None) is not None:`

Reply:
Hm, I suppose this is due to the move to `ClientSettings` right? I am wondering what is better, whether to limit this to client only or allow server to have this config as well.

But I am leaning towards keeping that as a Client only. Will think about it for a day and we will see.

Reply:
> I suppose this is due to the move to ClientSettings right?

Yes.

> I am wondering what is better, whether to limit this to client only or allow server to have this config as well.

In most cases, server only responds to the client requests so http session persistence is mostly needed by the client. The [Advanced Usage](https://docs.python-requests.org/en/latest/user/advanced/#advanced-usage) of the `requests.Session` also explains it from the perspective of the client.

Besides if the user still wants to use session persistence for the provider, it can be done like this.

```python
from oic.utils.settings import ServerSettings

server_settings = ServerSettings()
session = requests.Session()
session.verify = True
server_settings.request_session = session
```